### PR TITLE
[12.0][ADD] mis builder query drilldown

### PR DESCRIPTION
**Pull request by sergiocorato**
This PR add the ability to drilldown the values shown on the report.
Depends on https://github.com/OCA/mis-builder/pull/368

### CODE REVIEW

**Comment by AntoniRomera** [comment]
I have tested the module and it works correctly except for two things, to make the tests I have added the change you have made in the 12.0 branch directly to this module, I copy the example code to you. In V13 work fine

```
_inherit = "mis.report"



    def _fetch_queries(self, date_from, date_to, get_additional_query_filter=None):
        self.ensure_one()
        res = super()._fetch_queries(date_from, date_to, get_additional_query_filter)

        for query in self.query_ids:
            eval_context = {
                "env": self.env,
                "time": time,
                "datetime": datetime,
                "dateutil": dateutil,
                # deprecated
                "uid": self.env.uid,
                "context": self.env.context,
            }
            domain = query.domain and safe_eval(query.domain, eval_context) or []


            res.update({query: {"domain": domain}})

        return res
```

File: mis_builder_query_drilldown/models/mis_report.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+
+class MisQueryExpressionEvaluator(ExpressionEvaluator):
+    def eval_expressions(self, expressions, locals_dict):
+        vals, drilldown_args, name_error = super(
+            MisQueryExpressionEvaluator, self
+        ).eval_expressions(expressions, locals_dict)
+        for expression in expressions:
+            if expression.kpi_id.query_id:
+                domain = safe_eval(expression.kpi_id.query_id.domain)
```
Comment (AntoniRomera) [inline mis_builder_query_drilldown/models/mis_report.py]:
```suggestion
                domain = expression.kpi_id.query_id.domain and safe_eval(expression.kpi_id.query_id.domain) or []
```

File: mis_builder_query_drilldown/models/mis_report.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            no_auto_expand_accounts=self.no_auto_expand_accounts,
+        )
+
+    def drilldown(self, arg):
+        self.ensure_one()
+        period_id = arg.get("period_id")
+        domain = arg.get("domain", False)
+        model = arg.get("model", False)
+        date_field = arg.get("date_field", False)
+        if period_id and date_field and model and domain:
```
Comment (AntoniRomera) [inline mis_builder_query_drilldown/models/mis_report.py]:
```suggestion
        if period_id and date_field and model and isinstance(domain, list):
```

**Comment by sergiocorato** [comment]
Thanks @AntoniRomera. @OCA/mis-builder-maintainers: is anybody interested?

**Comment by AntoniRomera** [comment]
I think it's a good functionality, but I think it would be necessary to extend it to work with subkpi, but the module itself seems to me super useful.

**Pull request closed without merging**